### PR TITLE
Add flatpak-external-data-checker

### PR DESCRIPTION
**Pull request by gasinvein**
Why not distribute flatpak-external-data-checker as a flatpak?

### CODE REVIEW

**Comment by gasinvein** [comment]
bot, build org.flathub.flatpak-external-data-checker

**Comment by barthalion** [comment]
We could, but why? I submitted flat-manager-client and appstream-glib because that was the easiest way to get it working on our infrastructure, but external-data-checker is automated via GitHub actions, so glib and python versions are not a concern as Dockerfile is enough.

**Comment by gasinvein** [comment]
A flatpak would be easier to use locally than via Docker.

**Comment by gasinvein** [comment]
bot, build org.flathub.flatpak-external-data-checker

**Comment by gasinvein** [comment]
bot, build org.flathub.flatpak-external-data-checker

**Comment by gasinvein** [comment]
bot, build org.flathub.flatpak-external-data-checker

**Comment by barthalion** [comment]
bot, build org.flathub.flatpak-external-data-checker

File: org.flathub.flatpak-external-data-checker.yml [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+    buildsystem: simple
+    build-commands:
+      - mkdir -p /app/lib /app/bin
+      - cp -ax . /app/lib/fedc
+      - ln -sr /app/{lib/fedc,bin}/flatpak-external-data-checker
+      - ln -sr /app/{lib/fedc,bin}/canonicalize-manifest
+      - install -Dm644 ${FLATPAK_ID}.metainfo.xml -t /app/share/metainfo/
+    sources:
+      - type: git
+        url: "https://github.com/flathub/flatpak-external-data-checker.git"
```
Comment (bilelmoussaoui) [inline org.flathub.flatpak-external-data-checker.yml]: Use the latest available commit here for now.

File: org.flathub.flatpak-external-data-checker.metainfo.xml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="console-application">
+  <id>org.flathub.flatpak-external-data-checker</id>
+  <name>Flatpak External Data Checker</name>
+  <summary>A tool for checking if the external data used in Flatpak manifests is still up to date</summary>
+  <url type="homepage">https://github.com/flathub/flatpak-external-data-checker</url>
```
Comment (bilelmoussaoui) [inline org.flathub.flatpak-external-data-checker.metainfo.xml]: I would submit this upstream if possible.

File: flathub.json [hunk 3]
```diff
@@ -0,0 +1,3 @@
+{
+    "skip-appstream-check": true
```
Comment (bilelmoussaoui) [inline flathub.json]: Is this needed now that you're installing a metainfo?

Comment (gasinvein) [reply flathub.json]: The validator wants screenshots, but in this program there is really nothing to screenshot - it's non-interactive CLI.

Comment (bilelmoussaoui) [reply flathub.json]: https://github.com/flathub/org.freedesktop.appstream-glib/issues/9

File: org.flathub.flatpak-external-data-checker.yml (hunk 1 above)
Comment (gasinvein) [reply org.flathub.flatpak-external-data-checker.yml]: Done.

**Comment by LeandroStanger** [comment]
bot, build org.flathub.flatpak-external-data-checker

File: org.flathub.flatpak-external-data-checker.metainfo.xml (hunk 2 above)
Comment (gasinvein) [reply org.flathub.flatpak-external-data-checker.metainfo.xml]: Opened flathub/flatpak-external-data-checker#67

**Comment by LeandroStanger** [comment]
bot, build org.flathub.flatpak-external-data-checker

**Comment by barthalion** [comment]
Needs a bump to current master (there were some deps changed/added) and we can merge it if you're still interested.

**Comment by gasinvein** [comment]
@barthalion I see several new test-related dependencies there, but omitted them since the data-checker tests require network access, so we can't run them during flatpak build time. Is it correct?

bot, build org.flathub.flatpak-external-data-checker

**Comment by barthalion** [comment]
Right, I mostly meant defusedxml library but looks like it's included.

@wjt I don't expect you will actively maintain it but will add you to collaborators nonetheless.

/merge @wjt

**Pull request closed without merging**